eth/eventdatahandler/event_data_handler.go

```diff
@@ -276,22 +250,162 @@ func (edh *EventDataHandler) processEvent(txn eventdb.RW, event ethtypes.Log) (T
 			return nil, fmt.Errorf("handle FeeRecipientAddressUpdated: %w", err)
 		}
 
-		task := func() error {
-			if !updated {
-				return nil
-			}
+		if !updated {
+			return nil, fmt.Errorf("provided recipient address is the same")
+		}
+
+		task := NewTask(edh, event, nil)
+
+		return task, nil
+
+	default:
+		edh.logger.Warn("unknown event name", fields.Name(abiEvent.Name))
+		return nil, nil
+	}
+}
 
-			if err := edh.taskExecutor.UpdateFeeRecipient(feeRecipientAddressUpdatedEvent); err != nil {
-				return fmt.Errorf("update fee recipient: %w", err)
+// find and remove opposite tasks (start-stop, stop-start, liquidate-reactivate, reactivate-liquidate)
+// find superseding tasks and remove superseded ones (updateFee-updateFee)
+func cleanTaskList(tasks []*Task) []*Task {
```

Comment:
Let's remove it for now and add a TODO for cleaning tasks back. We'll discuss the logic later and then implement it, probably in the next PR.